traffic_ops/traffic_ops_golang/profile/profiles.go

```diff
@@ -149,7 +150,7 @@ func (prof *TOProfile) Read(h http.Header, useIMS bool) ([]interface{}, error, e
 
 	rows, err := prof.ReqInfo.Tx.NamedQuery(query, queryValues)
 	if err != nil {
-		return nil, nil, errors.New("profile read querying: " + err.Error()), http.StatusInternalServerError, nil
+		return nil, errors.New("bad request found"), errors.New("profile read querying: " + err.Error()), http.StatusBadRequest, nil
```

Comment:
So if you use `github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/api.ParseDBError` you can get possibly some more information that will help the client correct their request, rather than just telling them it was bad.

Reply:
I tried using ParseDBError function getting the same 500 Internal error. Instead, I updated the code to print the exact errors. It returns 400 Bad Request.